wlm/src/main/java/ai/djl/serving/wlm/LmiUtils.java

```diff
@@ -163,6 +208,76 @@ private static boolean isPythonDependencyInstalled(
         return false;
     }
 
+    private static Path buildTrtLlmArtifacts(Path modelDir, String modelId, String tpDegree)
+            throws IOException {
+        logger.info("Converting model to TensorRT-LLM artifacts");
+        String hash = Utils.hash(modelId + tpDegree);
+        String download = Utils.getenv("SERVING_DOWNLOAD_DIR", null);
+        Path parent = download == null ? Utils.getCacheDir() : Paths.get(download);
+        Path trtLlmRepoDir = parent.resolve("trtllm").resolve(hash);
+        if (Files.exists(trtLlmRepoDir)) {
+            logger.info("TensorRT-LLM artifacts already converted: {}", trtLlmRepoDir);
+            return trtLlmRepoDir;
+        }
+
+        String[] cmd = {
+            "python",
+            "/opt/djl/partition/trt_llm_partition.py",
+            "--properties_dir",
+            modelDir.toAbsolutePath().toString(),
+            "--trt_llm_model_repo",
+            trtLlmRepoDir.toString(),
+            "--tensor_parallel_degree",
+            tpDegree,
+            "--model_path",
+            modelId
+        };
+        boolean success = false;
+        try {
+            Process exec = new ProcessBuilder(cmd).redirectErrorStream(true).start();
+            try (BufferedReader reader =
+                    new BufferedReader(
+                            new InputStreamReader(exec.getInputStream(), StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    logger.info("convert_py: {}", line);
+                }
+            }
+            int exitCode = exec.waitFor();
+            if (0 != exitCode) {
+                throw new EngineException("Model conversion process failed!");
+            }
+            success = true;
+            logger.info("TensorRT-LLM artifacts built successfully");
+            return trtLlmRepoDir;
+        } catch (InterruptedException e) {
+            throw new IOException("Failed to build TensorRT-LLM artifacts", e);
+        } finally {
+            if (!success) {
+                Utils.deleteQuietly(trtLlmRepoDir);
+            }
+        }
+    }
+
+    static boolean isValidTrtLlmModelRepo(Path modelPath) throws IOException {
+        // TODO: match model name
+        AtomicBoolean isValid = new AtomicBoolean();
+        try (Stream<Path> walk = Files.list(modelPath)) {
+            walk.filter(Files::isDirectory)
+                    .forEach(
+                            p -> {
+                                Path confFile = p.resolve("config.pbtxt");
+                                // Path tokenizer = p.resolve("1/tokenizer_config.json");
```

Comment:
why is tokenizer check not included?

Reply:
tokenizer location has been changed, and it's not merged yet